lib/replica_pools/query_cache.rb

```diff
@@ -32,7 +34,7 @@ def select_all(*args)
       if raw_binds.blank? && relation.is_a?(ActiveRecord::Relation)
         arel, binds = relation.arel, relation.bind_values
       else
-        arel, binds = relation, raw_binds
+        arel, binds = relation, raw_binds.to_a
```

Comment:
Using `Array(raw_binds)` you prevent some `.to_a` "weaknesses" and keep the behavior you want, example:

```
Array(1) => [1]
1.to_a => NoMethodError: undefined method `to_a' for 1:Integer
```

Reply:
I've never used `Array` before, but now I know!
I think the binds variable is expected to be an array of variable bindings to the sql being executed, so I don't think we would expect value such as a single integer(??). I think this would be sufficient in the case of the `nil` case as we have seen with the `pluck` case. 

I took a look into the kernel method `Array` and found some people talking about it failing for hashes. Tried it out:

```

Reply:
Oh yeah, I did understand that we won't have a integer here, but it was just to show an example that using Array() instead of .to_a keeps the behavior and avoid possible issues as ruby is a untyped programming language.

I believe the main message here is that it's more assertive, you're ensuring it'll be returning an Array instead asking for some class to have a `.to_a` method.